### DISCUSSION TITLE
Can't deploy from main branch again?

### DISCUSSION THREAD

**Discussion by brianminges** [community/community#38957] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I tried to deploy a site but it posted the readme file instead. I then undeployed the site, but now GitHub won't let me deploy from my main branch again. The save function is grayed out. It will let me deploy from a branch, but I prefer not to do that. Any ideas how I can get that functionality back>?

**Comment by radar-blip** [comment]
Which repo is it? Are you missing an index.html?

Note: If the save function is grayed out, sounds like the main branch might still be in use. 
Are you sure you unpublished the site?

![unpublish](https://user-images.githubusercontent.com/39196767/201499450-1cf2c78d-c9a1-457d-a776-bf8593dd71f6.png)

**Reply by brianminges** [reply]
So it was unpublished. I just sent a new branch up and merged it, hoping that would solve the problem. What it did was make the page go live again. And what's worse, it's publishing the entire readme.md file instead of just my index. I have deployed to GH pages before and not run into this issue. What am I doing wrong? 

https://brianminges.github.io/calculator/

**Reply by radar-blip** [reply]
Its working, but only showing a blank page. 

In your index.html,  there is nothing in the body section except: `<div id="root"></div>`
Also, I dont see a link to the css.
I dont know exactly how javascript works (never liked it) but shouldn't there be more in the body section of the index.html?

**Reply by brianminges** [reply]
I moved the index.html to root level. The blank page showed up after I did that. It previously was in the public folder.

**Reply by brianminges** [reply]
> Its working, but only showing a blank page.
> 
> In your index.html, there is nothing in the body section except: `<div id="root"></div>` Also, I dont see a link to the css. I dont know exactly how javascript works (never liked it) but shouldn't there be more in the body section of the index.html?

It's made with React, so the id is being targeted in `src/index.js`

**Reply by radar-blip** [reply]
I dont know React, so I dont really know what errors to look for. 
Maybe someone who knows will chime in.

**Reply by rahmeh96** [reply]
Im having the same problem with react , any insight or solution?

**Comment by ghost** [comment]
hi guys sorry 
for me its SICK AND NEW..

**Reply by ghost** [reply]
helo guys
what saaup sorry i will try tomorrow to get more skill, wont make u get mad:D

**Comment by DSDmark** [comment, marked as answer]
Hi @brianminges,

So, It's also happened to me many times so, first if you not understand what to do then delete all your **github-workflows directory** and go to setting and in **environment section** delete all your pages and after that click on **page** section and deploy again and also note that it should look like this:-

![2022-11-13_10-23](https://user-images.githubusercontent.com/85744816/201506567-66f05253-f5be-4c15-b2a6-24d289d91708.png)

**Reply by ghost** [reply]
THANKS

**Comment by DSDmark** [comment]
Hi @brianminges,

use **gh-page** to deploy your React application. When you deploy react app, you need to build it first.

Read this for better understanding:- [**gh-page**](https://github.com/gitname/react-gh-pages)

And here is your git-page: [**your git page**](https://dsdmark.github.io/calculator/)

And here is your repo which I deploy, so it's help you to understand how you do deploy it, also don't forget to check **package.json** file.

[**Your forked repo**](https://github.com/DSDmark/calculator)
